package Exceptions;

import java.util.Scanner;

// Оператор throw (бросать)
/**
 * Чтобы сообщить о выполнении исключительных ситуаций в программе, можно
 * использовать оператор throw. То есть с помощью этого оператора мы сами можем
 * создать исключение и вызвать его в процессе выполнения. Например, в нашей
 * программе происходит ввод числа, и мы хотим, чтобы, если число больше 30, то
 * возникало исключение.
 * 
 * Здесь для создания объекта исключения используется конструктор класса
 * Exception, в который передается сообщение об исключении. И если число х
 * окажется больше 29, то будет выброшено исключение и управление перейдет к
 * блоку catch.
 * 
 * В блоке catch мы можем получить сообщение об исключении с помощью метода
 * getMessage().
 */
public class Program_03 {
    public static void main(String[] args) {
        try {
            try (Scanner in = new Scanner(System.in)) {
                System.out.println("Введите число меньше 30: ");
                int x = in.nextInt();
                if (x >= 30) {
                    throw new Exception("Число х должно быть меньше 30");
                }
            }
        } catch (Exception ex) {

            System.out.println(ex.getMessage());
        }
        System.out.println("Программа завершена");
    }
}
